app/utils/CheckYourAnswersHelper.scala

```diff
@@ -54,10 +52,7 @@ class CheckYourAnswersHelper(userAnswers: UserAnswers)(implicit messages: Messag
       )
   }
 
-  def displaySummaryFromInstruction(xml: Elem) : Seq[SummaryList.Row] = {
-    val arrangementID = (xml \ "ArrangementID").text
-    val importInstruction = (xml \ "DAC6Disclosures" \ "DisclosureImportInstruction").text
-    val disclosureID = (xml \ "DAC6Disclosures" \ "DisclosureID").text
+  def displaySummaryFromInstruction(importInstruction: String, arrangementID: Option[String], disclosureID: Option[String]) : Seq[SummaryList.Row] = {
```

Comment:
Why is this an option if you're unwrapping it further down in the method?

Reply:
I would have passed them as strings - however in CheckYourAnswerController when this method is called depending on the import instruction the arrangementID / DisclosureID may not be there. 

I could do a .getOrElse("") with an empty string when each of the values is extracted in the controller if you think this would be better.